lib/winston/transports/console.js

```diff
@@ -79,12 +80,24 @@ Console.prototype.log = function (info, callback) {
   // Remark: what if there is no raw...?
   //
   if (this.stderrLevels[info[LEVEL]]) {
-    process.stderr.write(info[MESSAGE] + this.eol);
+    if (console._stderr) {
+      // Node.js maps `process.stderr` to `console._stderr`.
+      console._stderr.write(info[MESSAGE] + this.eol);
+    } else {
+      console.error(info[MESSAGE] + this.eol);
+    }
+
     if (callback) { callback(); } // eslint-disable-line
     return;
   }
 
-  process.stdout.write(info[MESSAGE] + this.eol);
+  if (console._stdout) {
+    // Node.js maps `process.stdout` to `console._stdout`.
+    console._stdout.write(info[MESSAGE] + this.eol);
+  } else {
+    console.log(info[MESSAGE] + this.eol);
```

Comment:
this doesn't properly handle the `this.eol` it just appends `\n` to the message https://github.com/nodejs/node/blob/master/lib/console.js#L148. So we either shouldn't do this or we should remove `this.eol` from the message and accept the `\n`

Reply:
Good call. This targets browser support and `console.log` in the browser adds the newline for us so it should be dropped there.